server/aesdsocket.c

```diff
@@ -23,20 +23,52 @@
 #include <syslog.h>
 #include <getopt.h>
 #include <stdbool.h>
+#include "queue.h"
+#include <pthread.h>
+#include <time.h>
+#include <sys/time.h>
+
 
 #define PORT 9000
-#define MAX_CONNECTS 5
+#define MAX_CONNECTS 20
 #define FILE_NAME "/var/tmp/aesdsocketdata"
 #define BUFFER_SIZE 100
 #define LINE_SZ_TABLE_MAX 10000
+#define TIME_BUFFER_SIZE 150
+#define PRINT_BUFFER_SIZE 200
 
 
 // global variables
 struct sockaddr_in server_addr, client_addr;
 socklen_t addrlen;
-int server_fd, client_fd, fd;
-sigset_t intmask;
-char buffer[BUFFER_SIZE];
+int server_fd, fd;
+
+
+typedef struct thread_param_s thread_param_t;
+struct  thread_param_s
+{
+    int client_fd;
+    sigset_t intmask;  
+    bool complete;  
+    pthread_t thread_id;
+    pthread_mutex_t *mutex_lock;
+
+};
+
+// Setup linkedlist
+typedef struct slist_data_s slist_data_t;
+struct slist_data_s {
+
+    thread_param_t thread_param;
+    SLIST_ENTRY(slist_data_s) entries;
+};
+
+// mutex lock variable
+pthread_mutex_t mutex_lock;
```

Comment:
@SundarKrishnakumar think of a more descriptive name for this variable.  What is it locking?